source/common/router/config_impl.cc

```diff
@@ -261,26 +283,23 @@ const Route* RouteEntryImplBase::clusterEntry(uint64_t random_value) const {
       // sum(weights) > WeightedClusterEntry::MAX_CLUSTER_WEIGHT.
       // In this case, terminate the search and just return the cluster
       // whose weight cased the overflow
-      return cluster.get();
+      return cluster;
     }
     begin = end;
   }
   NOT_IMPLEMENTED;
 }
 
 void RouteEntryImplBase::validateClusters(Upstream::ClusterManager& cm) const {
-  // Throws an error if any of the clusters held by this Route or the
-  // internal weighted clusters are invalid.
   if (isRedirect()) {
     return;
   }
 
-  // We could have either a normal cluster or weighted cluster
-  if (weighted_clusters_.empty()) {
-    if (!cm.get(this->clusterName())) {
-      throw EnvoyException(fmt::format("route: unknown cluster '{}'", this->clusterName()));
+  if (!cluster_name_.empty()) {
+    if (!cm.get(cluster_name_)) {
```

Comment:
Yes, it will still error. My thinking on this was to leave the behavior as is for a statically defined routes. Theoretically I could see a case where someone might want to have a static route table, but then load the clusters from CDS, but this is an edge case. More common will be to use cluster_header. In the RDS case, I'm planning on disabling the checking so when a route table is loaded from RDS it will *not* assume that clusters already exist.

Basically, I would like to keep the checking functionality, at least behind an option, because it's useful for error checking static configs. IMO I would leave this as is for now and see how it looks when I do the RDS change next.